buildSrc/src/main/kotlin/datadog/gradle/plugin/naming/InstrumentationNamingExtension.kt

```diff
@@ -0,0 +1,43 @@
+package datadog.gradle.plugin.naming
+
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+
+/**
+ * Extension for configuring instrumentation naming convention checks.
+ *
+ * Example usage:
+ * ```
+ * instrumentationNaming {
+ *   instrumentationsDir.set(file("dd-java-agent/instrumentation"))
+ *   exclusions.set(listOf("http-url-connection", "sslsocket"))
+ *   suffixes.set(listOf("-common", "-stubs""))
+ * }
+ * ```
+ */
+abstract class InstrumentationNamingExtension {
+  /**
+   * The directory containing instrumentation modules.
+   * Defaults to "dd-java-agent/instrumentation".
+   */
+  abstract val instrumentationsDir: Property<String>
+
+  /**
+   * List of module names to exclude from naming convention checks.
+   * These modules will not be validated against the naming rules.
+   */
+  abstract val exclusions: ListProperty<String>
+
+  /**
+   * List of allowed suffixes for module names (e.g., "-common", "-stubs").
+   * Module names must end with either one of these suffixes or a version number.
+   * Defaults to ["-common", "-stubs"].
+   */
+  abstract val suffixes: ListProperty<String>
+
+  init {
+    instrumentationsDir.convention("dd-java-agent/instrumentation")
+    exclusions.convention(emptyList())
+    suffixes.convention(listOf("-common", "-stubs"))
+  }
+}
```

Comment:
I think this can be changed to follow the same pattern we have in other plugins:

```suggestion
abstract class InstrumentationNamingExtension @Inject constructor(objectFactory: ObjectFactory) {
  /**
   * The directory containing instrumentation modules.
   * Defaults to "dd-java-agent/instrumentation".
   */
   val instrumentationsDir: Property<String> = objectFactory.property(String::class.java).convention("dd-java-agent/instrumentation")

  /**
   * List of module names to exclude from naming convention checks.
   * These modules will not be validated against the naming rules.
   */
   val exclusions: ListProperty<String> = objectFactory.listProperty(String::class.java).convention(listOf("-common", "-stubs"))

  /**
   * List of allowed suffixes for module names (e.g., "-common", "-stubs").
   * Module names must end with either one of these suffixes or a version number.
   * Defaults to ["-common", "-stubs"].
   */
  val suffixes: ListProperty<String> = objectFactory.listProperty(String::class.java).convention(emptyList<String>())
}
```